format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/SplitQuotesIterator.java

```diff
@@ -94,4 +106,30 @@ protected String computeNext() {
 
     return split.toString();
   }
+
+
+  /**
+   * This method will fetch the next line from Record Reader if split is still within quotes which means data is also
+   * there in next line.
+   *
+   * @param isWithinQuotes whether current split value is within quotes or not
+   */
+  private void appendNextLineToDelimitedString(StringBuilder split, boolean isWithinQuotes) {
+    try {
+      if (enableMultilineSupport && index == delimitedString.length() && isWithinQuotes) {
+        if (recordReader.nextKeyValue()) {
```

Comment:
nit: in these situations it's a little more readable to check the error condition first:

```
if (!recordReader.nextKeyValue()) {
  throw ...
}
split.append(LINE_SEPARATOR);
...
```